twisted/test/test_text.py

```diff
@@ -181,66 +181,66 @@ def test_multiWord(self):
 
 class StrFileTests(unittest.TestCase):
     def setUp(self):
-        self.io = StringIO("this is a test string")
+        self.stream = StringIO(u"this is a test string")
```

Comment:
My suggestion is to use `BytesIO` only on Python 2.  Byte strings are strings on Python 2:

```python
print bytes is str # True
````

https://eval.in/612313

`io` is a back port from Python 3, so it has a confusing name for the file-like object that returns native strings on Python 2.  But `BytesIO` does return native strings on Python 2.

However, you should actually delete strFile.  It seems like it's is only used here:

https://github.com/twisted/twisted/blob/trunk/twisted/mail/imap4.py#L1638

And the tests for `imap4.py` use an implementation that returns a `cStringIO`/`StringIO` from `getBodyFile`:

https://github.com/twisted/twisted/blob/trunk/twisted/mail/test/test_imap.py#L3458-L3459

And `cStringIO`/`StringIO` returns native strings on Python 2, which are byte strings.  *However*, `strFile` should actually operate only on *bytes*, because that is what I'm sure all actual implementations of `IMailbox` and `IMessage` expect.   So `strFile` should go away and be replaced by `bytesFile` or something.  I don't know how doable that is, given that this is a public API.

Now, if you change all the tests to check `unicode` strings instead of whatever the native string type is in the running Python, the tests will no longer verify that public API.  Which makes the tests meaningless, and means they should be deleted, and with them `strFile`.

So: you can use my suggestion to get these tests passing on Python 2 and 3 in a way that doesn't involve implicit decoding somewhere, either in user code or in the tests.  Or you can delete `strFile` and replace it with something that will work correctly on Python 2 and 3 in the one place in Twisted where it's used.  But you should not change all these tests to search for Unicode strings in Unicode-aware file-like objects, because that is not what `strFile`'s API actually requires.